lesson3/Lesson3HW/app/src/main/java/com/margaret/lesson3hw/SettingsFragment.java

```diff
@@ -60,7 +50,8 @@ public void buttonSetup(Button name, final int color) {
         //change the background color on click
         name.setOnClickListener(new View.OnClickListener() {
             public void onClick(View v) {
-                getView().setBackgroundColor(color);
+                View activityView = getActivity().findViewById(R.id.mainLayout);
+                activityView.setBackgroundColor(color);
```

Comment:
In order to set the background color for the whole app (not just the current fragment), you need to give your entire main activity layout an id, and get it here.